Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -204,6 +204,11 @@ void RenderTheme::adjustStyle(RenderStyle& style, const Element* element, const
         style.setEffectiveAppearance(appearance);
     }
 
+    if (appearance == StyleAppearance::SearchField && searchFieldShouldAppearAsTextField(style)) {
+        appearance = StyleAppearance::TextField;
+        style.setEffectiveAppearance(appearance);
+    }
```

Comment:
I somewhat wished we had all of this logic in autoAppearanceForElement

Reply:
I wanted to keep the special case separate, similar to the `isControlStyled` cases. Going to leave this as-is unless you feel strongly.